tikv build error on macos

## Bug Report

<!-- Thanks for your bug report! Don't worry if you can't fill out all the sections. -->

### What version of TiKV are you using?
<!-- You can run `tikv-server --version` -->
```
[2022-01-08T14:58:43.802Z] error[E0605]: non-primitive cast: `NonZeroU64` as `u32`

[2022-01-08T14:58:43.802Z]    --> components/tikv_util/src/sys/thread.rs:210:9

[2022-01-08T14:58:43.802Z]     |

[2022-01-08T14:58:43.802Z] 210 |         std::thread::current().id().as_u64() as Pid

[2022-01-08T14:58:43.802Z]     |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ an `as` expression can only be used to convert between primitive types or to coerce to a specific trait object

[2022-01-08T14:58:43.802Z] 

[2022-01-08T14:58:43.802Z] error[E0308]: mismatched types

[2022-01-08T14:58:43.802Z]    --> components/tikv_util/src/sys/thread.rs:214:12

[2022-01-08T14:58:43.802Z]     |

[2022-01-08T14:58:43.802Z] 213 |     pub fn thread_ids<C: FromIterator<Pid>>(pid: Pid) -> io::Result<C> {

[2022-01-08T14:58:43.802Z]     |                       - this type parameter

[2022-01-08T14:58:43.802Z] 214 |         Ok(Some(thread_id()))

[2022-01-08T14:58:43.802Z]     |            ^^^^^^^^^^^^^^^^^ expected type parameter `C`, found enum `std::option::Option`

[2022-01-08T14:58:43.802Z]     |

[2022-01-08T14:58:43.802Z]     = note: expected type parameter `C`

[2022-01-08T14:58:43.802Z]                          found enum `std::option::Option<u32>`

[2022-01-08T14:58:43.802Z] 

[2022-01-08T14:58:43.802Z] error[E0609]: no field `stime` on type `ThreadStat`

[2022-01-08T14:58:43.802Z]    --> components/tikv_util/src/sys/thread.rs:235:27

[2022-01-08T14:58:43.802Z]     |

[2022-01-08T14:58:43.802Z] 235 |         s_time: full_stat.stime as _,

[2022-01-08T14:58:43.802Z]     |                           ^^^^^ help: a field with a similar name exists: `s_time`

[2022-01-08T14:58:43.802Z] 

[2022-01-08T14:58:43.802Z] error[E0609]: no field `utime` on type `ThreadStat`

[2022-01-08T14:58:43.802Z]    --> components/tikv_util/src/sys/thread.rs:236:27

[2022-01-08T14:58:43.802Z]     |

[2022-01-08T14:58:43.802Z] 236 |         u_time: full_stat.utime as _,

[2022-01-08T14:58:43.802Z]     |                           ^^^^^ help: a field with a similar name exists: `u_time`

[2022-01-08T14:58:43.802Z] 

[2022-01-08T14:58:44.062Z] Some errors have detailed explanations: E0308, E0605, E0609.

[2022-01-08T14:58:44.062Z] For more information about an error, try `rustc --explain E0308`.

[2022-01-08T14:58:44.062Z] error: could not compile `tikv_util` due to 4 previous errors
```

### What operating system and CPU are you using?
<!-- If you're using Linux, you can run `cat /proc/cpuinfo` -->

### Steps to reproduce
<!-- If possible, provide a recipe for reproducing the error. A complete runnable program is good. -->

### What did you expect?

### What did happened?

seems introduced by https://github.com/tikv/tikv/pull/11785
@BusyJay PTAL

--- a/components/resource_metering/src/recorder/mod.rs
+++ b/components/resource_metering/src/recorder/mod.rs
@@ -147,7 +147,7 @@ impl Recorder {
             // Clean up the data of the destroyed threads.
             if let Ok(ids) = thread::thread_ids::<HashSet<_>>(thread::process_id()) {
                 self.thread_stores.retain(|k, v| {
-                    let retain = ids.contains(&(*k as i32));
+                    let retain = ids.contains(&(*k as _));
                     assert!(retain || v.attached_tag.load().is_none());
                     retain
                 });

--- a/components/tikv_util/src/sys/thread.rs
+++ b/components/tikv_util/src/sys/thread.rs
@@ -187,7 +187,15 @@ mod imp {
     use std::iter::FromIterator;
 
     pub type Pid = u32;
-    pub type FullStat = super::ThreadStat;
+
+    #[derive(Default)]
+    pub struct FullStat {
+        // libc::clock_t is not used here because the definition of
+        // clock_t is different on linux and bsd.
+        pub stime: i64,
+        pub utime: i64,
+        pub command: String,
+    }
 
     lazy_static::lazy_static! {
         static ref PROCESS_ID: Pid = std::process::id();
@@ -207,11 +215,11 @@ mod imp {
     /// Gets the ID of the current thread.
     #[inline]
     pub fn thread_id() -> Pid {
-        std::thread::current().id().as_u64() as Pid
+        u64::from(std::thread::current().id().as_u64()) as Pid
     }
 
-    pub fn thread_ids<C: FromIterator<Pid>>(pid: Pid) -> io::Result<C> {
-        Ok(Some(thread_id()))
+    pub fn thread_ids<C: FromIterator<Pid>>(_pid: Pid) -> io::Result<C> {
+        Ok(std::iter::once(thread_id()).collect())
     }
 
     pub fn full_thread_stat(_pid: Pid, _tid: Pid) -> io::Result<FullStat> {

--- a/components/tikv_util/src/sys/thread.rs
+++ b/components/tikv_util/src/sys/thread.rs
@@ -190,8 +190,6 @@ mod imp {
 
     #[derive(Default)]
     pub struct FullStat {
-        // libc::clock_t is not used here because the definition of
-        // clock_t is different on linux and bsd.
         pub stime: i64,
         pub utime: i64,
         pub command: String,

